helm/hpcc/templates/dafilesrv.yaml

```diff
@@ -101,5 +102,45 @@ kind: ConfigMap
 {{ include "hpcc.addService" ( dict "root" $ "name" .name "service" $service "selector" .name "defaultPort" 7600 ) }}
 ---
 {{ include "hpcc.addCertificate" (dict "root" $ "name" .name "service" .service "component" "dafilesrv" "visibility" .service.visibility) }}
+{{- if eq .application "spray" -}}
+ {{- if $networkPolicyCtx.firstSprayService -}}
+  {{- $_ := set $networkPolicyCtx "dfuServerMap" dict -}}
+  {{- $_ := set $networkPolicyCtx "untargetedDfuServers" list -}}
+  {{- range $dfuServer := $.Values.dfuserver -}}
+   {{- if not $dfuServer.disabled -}}
+    {{- if hasKey $dfuServer "sprayServiceName" -}}
+     {{- $sprayServiceNames := (get $networkPolicyCtx.dfuServerMap $dfuServer.sprayServiceName) | default list -}}
+     {{- $sprayServiceNames := append $sprayServiceNames $dfuServer.name -}}
+     {{- $_ := set $networkPolicyCtx.dfuServerMap $dfuServer.sprayServiceName $sprayServiceNames -}}
+    {{- else -}}
+     {{- $_ := set $networkPolicyCtx "untargetedDfuServers" (append $networkPolicyCtx.untargetedDfuServers $dfuServer.name) -}}
+    {{- end -}}
+   {{- end -}}
+  {{- end -}}
+ {{ end }}
+ {{- $_ := set $commonCtx "allowedDfuServers" (ternary $networkPolicyCtx.untargetedDfuServers list $networkPolicyCtx.firstSprayService) -}}
```

Comment:
A comment above stating that dfu servers without a "sprayServiceName" set are assigned to the first spray service may help here.